handlers/sf-emails-to-s3-exporter/src/main/scala/com/gu/sf_emails_to_s3_exporter/Handler.scala

```diff
@@ -16,23 +16,78 @@ object Handler extends LazyLogging {
 
   def handleRequest(): Unit = {
 
-    val emailsFromSF = for {
+    for {
       config <- Config.fromEnvironment.toRight("Missing config value")
       authentication <- auth(config.sfConfig)
       sfAuthDetails <- decode[SfAuthDetails](authentication)
-      emailsFromSF <- getEmailsFromSfByQuery(sfAuthDetails, config.sfConfig.apiVersion)
-    } yield processEmails(sfAuthDetails, emailsFromSF, config.s3Config.bucketName)
+      asyncProcessRecs <- getRecordsFromSF[AsyncProcessRecsFromSfResponse.Response](
+        sfAuthDetails,
+        config.sfConfig.apiVersion,
+        GetAsyncProcessRecsQuery.query,
+        2000
+      )
+    } yield {
+      val asyncProcessRecIds = asyncProcessRecs.records.map(rec => rec.Id)
+
+      if (!asyncProcessRecIds.isEmpty) {
+
+        val batchedAsyncProcessRecs = batchAsyncProcessRecs(asyncProcessRecs.records, 200)
+
+        batchedAsyncProcessRecs.map(
+
+          asyncProcessRecGroup => {
```

Comment:
```suggestion
        batchedAsyncProcessRecs.map { asyncProcessRecGroup =>
```
and remove the closing `)` on line 56